src/sentry/api/endpoints/organization_member/details.py

```diff
@@ -196,32 +205,28 @@ def put(
                 # TODO(dcramer): proper error message
                 return Response({"detail": ERR_UNINVITABLE}, status=400)
 
-        if "teams" in result:
-            # dupe code from member_index
-            # ensure listed teams are real teams
-            teams = list(
-                Team.objects.filter(
-                    organization=organization, status=TeamStatus.VISIBLE, slug__in=result["teams"]
-                )
-            )
-
-            if len(set(result["teams"])) != len(teams):
-                return Response({"teams": "Invalid team"}, status=400)
-
-            with transaction.atomic():
-                # teams may be empty
-                OrganizationMemberTeam.objects.filter(organizationmember=member).delete()
-                OrganizationMemberTeam.objects.bulk_create(
-                    [OrganizationMemberTeam(team=team, organizationmember=member) for team in teams]
-                )
-
-        assigned_role = result.get("role")
-        if assigned_role:
+        # Set the team-role before org-role. If the org-role has elevated permissions
+        # on the teams, the team-roles can be overwritten later
+        try:
```

Comment:
We should enclose this whole block in
```python
if "teamRoles" in result or "teams" in result
```

It's costly to acquire the lock if we aren't going to use it.